### PR TITLE
fix: add callback validation for collections in dq (CMC-1562)

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-1562

### Change description ###

Add standard callback validation for witnesses and unavailable dates in DQs

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by pd-dudkiewicz — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/unspec/validation/interfaces/WitnessesValidator.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    default CallbackResponse validateRespondentDqWitnesses(CallbackParams callbackParams) {
+        return validateWitnesses(callbackParams.getCaseData().getRespondent1DQ());
+    }
+
+    default CallbackResponse validateApplicantDqWitnesses(CallbackParams callbackParams) {
+        return validateWitnesses(callbackParams.getCaseData().getApplicant1DQ());
+    }
+
+    private CallbackResponse validateWitnesses(DQ dq) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/validation/interfaces/WitnessesValidator.java:23]: probably we can just make this as default method and pass applicant or respondent DQ object in the calling classes. we don't need the above two default methods.

**Review by sabahirfan — APPROVED** [review]
one minor suggestions otherwise LGTM.

File: src/main/java/uk/gov/hmcts/reform/unspec/validation/interfaces/WitnessesValidator.java (hunk 1 above)
Comment (swalker125) [reply src/main/java/uk/gov/hmcts/reform/unspec/validation/interfaces/WitnessesValidator.java:23]: Good idea, I'm going to raise as a tech debt, we can probably do the same for the Experts

Comment (swalker125) [reply src/main/java/uk/gov/hmcts/reform/unspec/validation/interfaces/WitnessesValidator.java:23]: Made here: https://tools.hmcts.net/jira/browse/CMC-1568

**Pull request merged**